tests/infra/remote_shim.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import infra.remote
+import docker
+import re
+import os
+import pathlib
+import grp
+import ipaddress
+import infra.github
+
+
+from loguru import logger as LOG
+
+
+def is_docker_env():
+    """Returns true if the process executing _this_ code already runs inside Docker"""
+    return os.path.isfile("/.dockerenv")
+
+
+def is_azure_devops_env():
+    return "SYSTEM_TEAMFOUNDATIONCOLLECTIONURI" in os.environ
+
+
+def map_azure_devops_docker_workspace_dir(workspace_dir):
+    return workspace_dir.replace("__w", "/mnt/vss/_work")
+
+
+# Docker image name prefix
+DOCKER_IMAGE_NAME_PREFIX = "ccfciteam/ccf-app-run"
+
+# Network name
+AZURE_DEVOPS_CONTAINER_NETWORK_ENV_VAR = "AGENT_CONTAINERNETWORK"
+DOCKER_NETWORK_NAME_LOCAL = "ccf_test_docker_network"
+
+# Identifier for all CCF test containers
+CCF_TEST_CONTAINERS_LABEL = "ccf_test"
+
+
+def kernel_has_sgx_builtin():
+    with open("/proc/cpuinfo", "r", encoding="utf-8") as cpu_info:
+        f = re.compile("^flags.*sgx.*")
+        for line in cpu_info:
+            if f.match(line):
+                return True
+    return False
+
+
+class PassThroughShim(infra.remote.CCFRemote):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+# Current limitations, which should be overcomable:
+# 1. No support for SGX kernel built-in support (i.e. 5.11+ kernel) in Docker environment
+# 2. No support for concurrent test: all nodes containers attach to the same network at pre-defined IPs
```

Comment:
Current limitations are listed here, but I'd like to address them in a separate PR.